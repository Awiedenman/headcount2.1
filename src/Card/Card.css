.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 20px;
  width: 190px;
  height: 285px;
  box-shadow: 5 5 5 5px;
  border: 1px #000 solid;
  box-shadow: 1px 1px 1px 1px;
  background-color: lavenderblush
}

h1 {
  font-size: 20px;
  text-align: center;
}

p {
  font-size: 12px;
  margin: 1px
}

.under {
  color: lightgreen;
}

.over {
  color: purple;
}

.clicked {
  border: magenta 5px solid;
}