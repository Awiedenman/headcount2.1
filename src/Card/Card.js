import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

export const Card = ( {location, stats, addClickedCard, clicked } ) => {
  const districtStats = Object.keys(stats).map((year, index) => {
    return (
      <li 
        className={stats[year] < .5 ? 'under' : 'over'} 
        key={ index }> {`${ year } : ${ stats[year] }`}
      </li>
    );
  });


  return (
    <div 
      className={`card ${clicked ? 'clicked' : ''}`}  
      onClick={ ()=> addClickedCard( location ) }>
      <h1>{ location }</h1>
      <ul>{ districtStats }</ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string,
  stats: PropTypes.object,
  addClickedCard: PropTypes.func,
  clicked: PropTypes.bool
};