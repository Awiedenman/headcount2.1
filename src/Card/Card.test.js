import React from 'react';
import { Card } from './Card';
import { shallow, mount } from 'enzyme';

describe('Card', () => {
  
    const mockData = {location: 'Colorado', stats : {2004: 0.24, 2005: 0.278, 2006: 0.337, 2007: 0.395}};

  it('matches a Snapshot of the Card', () => {
    const wrapper = shallow(
      <Card
        key={1}
        location={'Colorado'}
        stats={ mockData }
      />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have classname of "clicked" if prop clicked is true', () => {
    const wrapper = shallow(<Card
        key={1}
        location={'Colorado'}
        stats={ mockData }
        clicked={true}
      />);
    expect(wrapper.is('.clicked')).toEqual(true);
  })

  it('should not have a classname of "clicked" if prop clicked is false', () => {
    const wrapper = shallow(<Card
        key={1}
        location={'Colorado'}
        stats={ mockData }
        clicked={false}
      />);
    expect(wrapper.is('.clicked')).toEqual(false);
  })

  it.skip('li in Card should have a classname of over if value is greater than .5', () => {
    const mockData1 = {location: 'Colorado', stats : {2004: 0.789, 2005: 0.342, 2006: 0.111, 2007: 0.395}};
    const wrapper = shallow(<Card
        key={1}
        location={'Colorado'}
        stats={ mockData1 }
        clicked={false}
        />);
    console.log(wrapper.find('.over'))
    expect(wrapper.find('.over').render()).toEqual('2004: 0.789');

  })

  it('should call addClickedCard method when card is clicked', () => {
    const mockAddClickedCard = jest.fn();
    const wrapper = shallow(<Card
        key={1}
        location={'Colorado'}
        stats={ mockData }
        clicked={false}
        addClickedCard={mockAddClickedCard}
      />);
    wrapper.simulate('click');
    expect(mockAddClickedCard).toBeCalled();
  })

});