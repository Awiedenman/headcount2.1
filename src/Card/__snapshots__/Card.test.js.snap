// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches a Snapshot of the Card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    location="Colorado"
    stats={
        Object {
            "location": "Colorado",
            "stats": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Colorado
</h1>,
        Array [
          <p
            className="over"
>
            location
             : 
            Colorado
</p>,
          <p
            className="over"
>
            stats
             : 
            Object {
                        "2004": 0.24,
                        "2005": 0.278,
                        "2006": 0.337,
                        "2007": 0.395,
                      }
</p>,
        ],
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Colorado",
        },
        "ref": null,
        "rendered": "Colorado",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "location",
            " : ",
            "Colorado",
          ],
          "className": "over",
        },
        "ref": null,
        "rendered": Array [
          "location",
          " : ",
          "Colorado",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "stats",
            " : ",
            Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
            },
          ],
          "className": "over",
        },
        "ref": null,
        "rendered": Array [
          "stats",
          " : ",
          Object {
            "2004": 0.24,
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Colorado
</h1>,
          Array [
            <p
              className="over"
>
              location
               : 
              Colorado
</p>,
            <p
              className="over"
>
              stats
               : 
              Object {
                            "2004": 0.24,
                            "2005": 0.278,
                            "2006": 0.337,
                            "2007": 0.395,
                          }
</p>,
          ],
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "location",
              " : ",
              "Colorado",
            ],
            "className": "over",
          },
          "ref": null,
          "rendered": Array [
            "location",
            " : ",
            "Colorado",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "stats",
              " : ",
              Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
              },
            ],
            "className": "over",
          },
          "ref": null,
          "rendered": Array [
            "stats",
            " : ",
            Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
